jupyterhub/traitlets.py

```diff
@@ -62,11 +62,14 @@ def validate(self, obj, value):
         If it has one of the suffixes, it is converted into the appropriate
         pure byte value.
         """
-        if isinstance(value, int):
-            return value
-        num = value[:-1]
+        if isinstance(value, float) or isinstance(value, int):
+            return float(value)
+        try:
+            num = float(value[:-1])
+        except ValueError:
+            raise TraitError('{val} is not a valid memory specification. Must be an float or a string with suffix K, M, G, T'.format(val=value))
         suffix = value[-1]
-        if not num.isdigit() and suffix not in ByteSpecification.UNIT_SUFFIXES:
-            raise TraitError('{val} is not a valid memory specification. Must be an int or a string with suffix K, M, G, T'.format(val=value))
+        if suffix not in ByteSpecification.UNIT_SUFFIXES:
+            raise TraitError('{val} is not a valid memory specification. Must be an float or a string with suffix K, M, G, T'.format(val=value))
         else:
-            return int(num) * ByteSpecification.UNIT_SUFFIXES[suffix]
+            return float(num) * ByteSpecification.UNIT_SUFFIXES[suffix]
```

Comment:
Shouldn't this be `int(float(num) * ByteSpec...)` to cast it back to int after scaling?